addOns/spider/src/main/java/org/zaproxy/addon/spider/SpiderTask.java

```diff
@@ -113,7 +113,7 @@ public SpiderTask(Spider parent, SpiderResourceFound resourceFound, URI uri) {
                         resourceFound.getMessage().getRequestHeader().getURI().toString());
             }
             HttpMessage msg = new HttpMessage(requestHeader);
-            if (!resourceFound.getBody().isEmpty()) {
+            if (!"GET".equals(resourceFound.getMethod())) {
```

Comment:
- `HttpRequestHeader.GET`
- It should set content-length if a GET with body though (unlikely to happen with the spider but since we don't disallow that when creating the resources found we should not send invalid requests, e.g. logic like `AbstractPlugin.updateRequestContentLength` would be more appropriate).

Reply:
Thanks